federation/pkg/federation-controller/service/dns.go

```diff
@@ -263,6 +266,9 @@ func (s *ServiceController) ensureDnsRecords(clusterName string, cachedService *
 	serviceName := cachedService.lastState.Name
 	namespaceName := cachedService.lastState.Namespace
 	zoneNames, regionName, err := s.getClusterZoneNames(clusterName)
+	if zoneNames == nil {
+		return fmt.Errorf("fail to get cluster zone names")
```

Comment:
Fair enough.  That's a bug in getClusterZoneNames.  So as not to hold you up, lets merge as is, and I'll fix that in a followup PR.